Update readme with more information

neurosnap opened the pull request:
I think a **Why** section would be useful for the documentation because it would explain the benefits of not using JSX with react.

An **API** section helped me understand all the different configurations in a clear and concise manner.

Jador commented on README.md:
```diff
@@ -1,13 +1,60 @@
+* No need for a JSX syntax highlighter
+* No need for a JSX linter
+* JSX elements are just functions anyway, why try to hide what it is?
+* HTML markup is ugly
```
too subjective

Jador commented:
Thanks a lot for this, especially the api section. There was one little thing I took exception with, which I commented on. But then I'll be happy to accept this PR.

Jador commented:
I just noticed that most of this comes from the readme for your project, which is perfectly fine with me. I would like it if you brought over the alternatives section too.

neurosnap commented:
My project was more of a way for me to figure out how I would approach dealing with `React.createElement`, I don't really plan on publishing it because it's virtually identical to this library.

neurosnap commented:
What part of the Why section is too subjective?  All of it or just the comment about HTML markup being ugly?

Jador commented:
The part about the HTML (not that I don't agree lol)

neurosnap commented:
I updated the **Pros** section to remove the subjective comments.

I also added an **Alternatives** section.

What do you think about coming up with some examples to illustrate the differences between JSX, `React.createElement`, and rhh in the README?

Jador commented:
Sure, sounds like a good idea

On Sun, Apr 17, 2016, 4:41 PM Eric Bower <notifications@github.com> wrote:

> I updated the *Pros* section to remove the subjective comments.
>
> I also added an *Alternatives* section.
>
> What do you think about coming up with some examples to illustrate the
> differences between JSX, React.createElement, and rhh in the README?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/Jador/react-hyperscript-helpers/pull/12#issuecomment-211120331>
>

Jador commented:
@neurosnap, I know if you're like me you probably don't have a ton of time to devote to this. If you like I can go ahead and accept this PR now and you can make another for the api differences later.

neurosnap commented:
I'd prefer to send a separate PR for that, but I am working on it.  Any suggestions for a good example?  Right now I'm using redux todo as my template.
